File: Nth catalan number - GFG/nth-catalan-number.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution
{
    public:
    //Function to find the nth catalan number.
    const int m = 1000000007;
    
    int findCatalan(int n) 
    {
        long long dp[n+1];
    dp[0] = 1; dp[1] = 1;
    for (int i = 2; i <= n; i++) {
        dp[i] = 0;
        for (int j = 0; j < i; j++) {
            dp[i] = (dp[i] + (dp[j] * dp[i - j - 1]) % m) % m;
        }
    }
    return (int)dp[n];
    }
};

//{ Driver Code Starts.
int main() 
{
	int t;
	cin>>t;
	while(t--) {
	    
	    int n;
	    cin>>n;
	    Solution obj;
	    cout<< obj.findCatalan(n) <<"\n";    
	}
	return 0;
}
// } Driver Code Ends

File: Nth catalan number - GFG/nth-catalan-number.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;

class GFG {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t;
        t = Integer.parseInt(br.readLine());
        while(t-- > 0){
            
            int n;
            n = Integer.parseInt(br.readLine());
            
            Solution obj = new Solution();
            int res = obj.findCatalan(n);
            
            System.out.println(res);
            
        }
    }
}

// } Driver Code Ends



class Solution {
   static int m = 1000000007;
    public static int findCatalan(int n) {
        long dp[] = new long[n+1];
        dp[0]=1; dp[1]=1;
        for(int i =2; i<=n ; i++)
        {
            for(int j=0; j<i; j++)
            dp[i] = (dp[i]+ (dp[j]*dp[i-j-1])%m)%m;
        }
        return (int)dp[n];
    }
}
  
        
